src/main.ts

```diff
@@ -37,16 +39,49 @@ export class Minifier {
 
   // Recursively visits every child node, emitting text of the sourcefile that is not a part of
   // a child node.
-  visit(node: ts.Node) {
+  visit(node: ts.Node, typeChecker?: ts.TypeChecker) {
     switch (node.kind) {
-      case ts.SyntaxKind.PropertyAccessExpression: {
-        let output = '';
-        let propAccessExp = <ts.PropertyAccessExpression>node;
-        output += this.visit(propAccessExp.expression);
-        output += '.';
-        // Adds '$mangled' to an identifier to ensure that the identifier is being altered.
-        output += propAccessExp.name.text + '$mangled';
-        return output;
+      case ts.SyntaxKind.Identifier: {
+        let parent = node.parent;
+
+        if (parent.kind === ts.SyntaxKind.PropertyDeclaration) {
+          return this.renameIdent(node);
+        } else if (parent.kind === ts.SyntaxKind.PropertyAccessExpression) {
+          let pae = <ts.PropertyAccessExpression>parent;
+          let exprSymbol = typeChecker.getTypeAtLocation(pae.expression).symbol;
+          if (!exprSymbol) {
+            exprSymbol = typeChecker.getSymbolAtLocation(pae.expression);
+          }
+          if (!exprSymbol) {
+            exprSymbol = typeChecker.getSymbolAtLocation(pae);
+          }
+          let childText = '';
+          if (exprSymbol) {
+            // start off by assuming the property is rename-able
+            let rename: boolean = true;
+
+            // check if a source filename of a declaration ends in .d.ts
+            exprSymbol.declarations.forEach((decl) => {
+              let fileName = decl.getSourceFile().fileName;
+              if (fileName.match(/\.d\.ts/)) {
+                rename = false;  // we can no longer rename the property
+              }
+            });
+
+            if (rename) {
+              childText = this.renameIdent(node);
+            } else {
+              childText = this.ident(node);
+            }
+          } else {
+            childText = this.renameIdent(node);
```

Comment:
Isn't this branch dangerous somehow? We found a property use site, but we have no idea where it's coming from, could be external or internal? If we guess incorrectly, we could break the program. Should this rather report an error?